### ISSUE TITLE
Argument to specify the directory for *.txt files created by data.utils.autosplit

### ISSUE THREAD

**Issue by austinhnelson** [ultralytics/ultralytics#20122]
### Search before asking

- [x] I have searched the Ultralytics [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar feature requests.

### Description

Hey there, I recently used the autosplit util function and found it to be quite useful in my specific use case. However, I would like to add the ability to specify the output directory for the txt files that get generated from autosplit. Currently, it just places them one directory above the image path that's passed into the function call.

### Use case

I'm running a dockerized environment and trying to keep all my paths/files nicely organized and not being able to specify where these txt files are put disrupts the pattern I'm going for. If I can put them in the same directory my data.yaml is, it keeps my project structure nicer.

### Are you willing to submit a PR?

- [x] Yes I'd like to help by submitting a PR!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @austinhnelson, thank you for your interest in Ultralytics 🚀 and for the detailed feature request! We appreciate your suggestion regarding the ability to specify an output directory for the `autosplit` function. This sounds like a great idea to enhance usability and maintain better project structure, especially in environments like Docker.

If you're planning to submit a PR, that's fantastic! 🎉 Please ensure you're following our [Contributing Guidelines](https://github.com/ultralytics/ultralytics/blob/main/CONTRIBUTING.md). This will help streamline the process and get your contributions reviewed more efficiently.

In the meantime, here are a few resources you might find helpful:
- Check out the [Docs](https://docs.ultralytics.com/) for more details on our utilities and features.
- Join the Ultralytics community for discussions and support. You can connect on [Discord](https://discord.com/invite/ultralytics) 🎧, [Discourse](https://community.ultralytics.com/), or our [Subreddit](https://reddit.com/r/Ultralytics).

## Upgrade
Before submitting your PR or continuing with development, make sure you're using the latest version of the `ultralytics` package. You can upgrade using:
```bash
pip install -U ultralytics
```

## Environments
For testing your changes, you can utilize verified environments with all dependencies preinstalled:
- **Notebooks**: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud**: See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon**: See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker**: See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## CI Status
<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>
If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are passing. CI tests verify correct operation of YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu.

This is an automated response to ensure you have all the information you need to get started. An Ultralytics engineer will review and assist you with your request soon. Thank you for contributing to the Ultralytics community! 😊

**Comment by glenn-jocher** [comment]
Hi @austinhnelson,

Thanks for the feature request! This would indeed be a useful addition to the `autosplit` function. Adding an optional parameter to specify the output directory for the text files makes sense, especially in containerized environments where path management is important.

Since you're willing to submit a PR, I'd suggest:

1. Add an optional parameter like `output_dir=None` to the function
2. Keep the current behavior as the default (when `output_dir=None`)
3. When specified, use the custom path for writing the `autosplit_*.txt` files

This should be a straightforward enhancement that maintains backward compatibility. Let me know if you need any guidance during implementation!

**Comment by austinhnelson** [comment]
Hey @glenn-jocher, thanks for those suggestions!

Also, I've discovered that if there aren't enough images to satisfy the weights specified by the autosplit function, no .txt file is generated. Even if I manually create these .txt files, the call to `model.train()` will fail because the .txt files are empty so it fails to parse.

I don't really think this is in scope of this PR but is there anyway to utilize autosplit and have training work, even if there aren't enough images to fulfill the weights? 

To give background on why I'm asking this, I have a very simple system level test that I'm testing a pipeline with and YOLO model training is a part of it. For testing, I don't want to pull a lot of data, preferably just one image/label. But with autosplit, I don't have enough data to fulfill the weights for test / val.

**Comment by glenn-jocher** [comment]
Hi @austinhnelson,

That's a good point about small datasets - I can see how this would be an issue when running tests with minimal data. 

For your immediate testing needs, you could manually create `autosplit_train.txt` and place your single image path there, skipping the autosplit function entirely. But I agree this would be a useful enhancement to handle automatically.

When implementing the `output_dir` parameter, you could also add logic to ensure that even with insufficient images, the function still creates valid (though potentially empty) files for val/test splits, and guarantees at least one image in the train split. This would make autosplit more robust for small datasets and testing scenarios.

If you'd like to include this enhancement in your PR alongside the output directory feature, that would be a welcome improvement!
